### PR TITLE
Ensure font-face styles are printed in iframe editors.

### PR DESCRIPTION
**Pull request by hellofromtonya**
## What?
In the plugin, ensure `wp_print_font_faces()` is invoked and added to the iframed editors' styles.

## Why?
Gutenberg overrides Core's `_wp_get_iframed_editor_assets()` styles results. Font Face's `wp_print_font_faces()` is in Core's `_wp_get_iframed_editor_assets()`. But Gutenberg overrides the results of that function in `_gutenberg_get_iframed_editor_assets()` (for WP 6.3) and `_gutenberg_get_iframed_editor_assets_6_4()` (for WP 6.4).

## How?

Loads the plugin's `fonts.php` file and moves the hooked code for `'block_editor_settings_all'` out of the guard.

## Testing Instructions

1. Set `FONT_LIBRARY_ENABLE` constant to `true` in your test site's `config.php`:
```php
define( 'FONT_LIBRARY_ENABLE', true );
```
2. Activate the Gutenberg plugin.
3. Go to the Site Editor.
4. Open your browser's dev tools and search  the HTML for `wp-fonts-local`.

Expected: After applying this PR, it should also exist in the `iframe`.

### CODE REVIEW

**Comment by ironprogrammer** [comment]
Thanks for the PR!

## Test Report

### Steps to Test
1. Configure environment to use `wordpress-develop:trunk` (current 6.4-alpha branch).
2. In *Appearance > Editor*, navigate to *Styles > Typography > Headings* and set the font to IBM Plex Mono. Then Save the global styles settings.
3. Observe that the headings display with the fallback face `monospace`. (This fallback may appear differently depending on your system. If unsure, use the browser inspector to override an element with `font-family: monospace` and note that they visually match.)
4. Open the browser inspector and observe that the editor's iframe does not include `<style id="wp-fonts-local">` element in the document's `<head>`. The theme's `@font-face` definitions are missing.
5. Apply patch/check out PR.
6. Refresh the editor.
7. Observe that the headings display in the correct font. (Also refer to the [IBM Plex Mono specimen page](https://fonts.google.com/specimen/IBM+Plex+Mono) for visual confirmation.)
8. Observe that the `<style id="wp-fonts-local">` element exists in the document's `<head>`. The tag includes `@font-face` definitions for active fonts.

### Environment
- OS: macOS 13.5.1
- Browser: Safari 16.6
- PHP: 8.2.10
- WordPress: 6.4-alpha-56267-src
- Theme: twentytwentythree v1.2
- Active Plugins:
	- gutenberg v16.6.0

### Actual Results
**Reproduce Issue**
- ✅ Fallback to `monospace` issue reproduced, *Figure 1*.
- ✅ The `#wp-fonts-local` element is missing from the iframed document's `<head>`.

**Test Patch**
- ✅ Heading displays with correct font face after patch, *Figure 2*.
- ✅ The `<head>` in the iframe includes the `#wp-fonts-local` element, and includes the `@font-face` definitions.

### Supplemental Artifacts
Figure 1 - before patch:
heading in monospace

Figure 2 - after patch:
heading in IBM Plex Mono

File: lib/compat/wordpress-6.4/fonts/fonts.php [hunk 1]
```diff
@@ -34,20 +34,20 @@ function wp_print_font_faces( $fonts = array() ) {
...
-			ob_start();
-			// @core-merge: add only this line into Core's `_wp_get_iframed_editor_assets()` function after `wp_print_styles()`.
-			wp_print_font_faces();
-			$styles = ob_get_clean();
+// @core-merge: do not merge this code into Core.
+add_filter(
+	'block_editor_settings_all',
+	static function( $settings ) {
+		ob_start();
+		// @core-merge: add only this line into Core's `_wp_get_iframed_editor_assets()` function after `wp_print_styles()`.
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.4/fonts/fonts.php:44]:
```suggestion
```
Should this comment be removed, since `wp_print_font_faces()` is already in core as of commit [c29b095](https://github.com/ironprogrammer/wordpress-develop/commit/c29b095e762836e47df9080d670c2f6570d6cd12#diff-edfc853e670c7bfc5a838dc74f75ba5d8b95cb4adbddf0644d9227bcb692e344R364)?

**Review by ironprogrammer — APPROVED** [review]
This looks good to me. I've got one small comment question, but it's not worth holding things up, as it'll be obvious when merging to Core.

File: lib/compat/wordpress-6.4/fonts/fonts.php (hunk 1 above)
Comment (anton-vlasenko) [reply lib/compat/wordpress-6.4/fonts/fonts.php:44]: cc @hellofromtonya

**Comment by anton-vlasenko** [comment]
## Test Report

### Environment
- **WordPress:** 6.4-alpha-56267-src
- **PHP:** 7.3.33
- **Server:** Apache/2.4.57 (Unix) PHP/7.3.33
- **Database:** mysqli (Server: 5.7.43)
- **Browser:** Safari 16.6 (macOS)
- **Theme:** Twenty Twenty-Three 1.2
- **MU-Plugins:** None activated
- **Plugins:** 
  - Gutenberg

### Steps to Test
1. Activate Gutenberg.
2. Check out the trunk branch.
3. Navigate to the Site Editor.
4. Open your browser's dev tools and search the HTML for `#wp-fonts-local` elements.
5. Note the number of `#wp-fonts-local` elements.
6. Check out this PR.
7. Navigate to the Site Editor again.
8. Open your browser's dev tools and search the HTML for `#wp-fonts-local` elements.
9. Note the number of `#wp-fonts-local` elements.

### Expected Results
- ✅ After applying this PR, `#wp-fonts-local` should exist in the iframe.

### Actual Results
- ❌ With Gutenberg enabled, I observe a similar number of `#wp-fonts-local` elements regardless of whether I'm on the trunk branch or have checked out this PR. In both scenarios (i.e. step `#5` vs step `#9`), I see 2 `#wp-fonts-local` elements in the root `<head>` element and 12 `#wp-fonts-local` elements within iframe elements.
In other words, the changes in this PR do not seem to affect the quantity of `#wp-fonts-local` elements.

File: lib/compat/wordpress-6.4/fonts/fonts.php (hunk 1 above)
Comment (hellofromtonya) [reply lib/compat/wordpress-6.4/fonts/fonts.php:44]: >Should this comment be removed, since `wp_print_font_faces()` is already in core as of commit [c29b095](https://github.com/ironprogrammer/wordpress-develop/commit/c29b095e762836e47df9080d670c2f6570d6cd12#diff-edfc853e670c7bfc5a838dc74f75ba5d8b95cb4adbddf0644d9227bcb692e344R364)?

It is. However, the intent is to retain these merge notes for history as the compat files relate directly to the WP version.

**Comment by hellofromtonya** [comment]
@anton-vlasenko did you set the `FONT_LIBRARY_ENABLE` constant in your test site's `wp-config.php` file?

```php
define( 'FONT_LIBRARY_ENABLE', true );
```

This constant turns on the Font Library and Font Face code in the plugin and turns off the Fonts API.

**Comment by anton-vlasenko** [comment]
## Test Report

### Environment
- **WordPress:** 6.4-alpha-56267-src
- **PHP:** 7.3.33
- **Server:** Apache/2.4.57 (Unix) PHP/7.3.33
- **Database:** mysqli (Server: 5.7.43)
- **Browser:** Safari 16.6 (macOS)
- **Theme:** Twenty Twenty-Three 1.2
- **MU-Plugins:** None activated
- **Plugins:** 
  - Gutenberg

### Steps to Test
1. Set the `FONT_LIBRARY_ENABLE` constant to `true` in your test site's `config.php`.
2. Activate Gutenberg.
3. Check out the trunk branch.
4. Navigate to the Site Editor.
5. Open your browser's dev tools and search the HTML for `#wp-fonts-local` elements.
6. Note the number of `#wp-fonts-local` elements.
7. Check out this PR.
8. Navigate to the Site Editor again.
9. Open your browser's dev tools and search the HTML for `#wp-fonts-local` elements.
10. Note the number of `#wp-fonts-local` elements.

### Expected Results
- ✅ After applying this PR, `#wp-fonts-local` should exist in the iframe.

### Actual Results
- ✅ After applying this PR, `#wp-fonts-local` elements are present in the frame elements on the page. 
There were no `#wp-fonts-local` elements in the iframe elements before applying this patch.

**Review by anton-vlasenko — APPROVED** [review]
After retesting the PR with the `FONT_LIBRARY_ENABLE` constant set to enabled, it functions as intended: https://github.com/WordPress/gutenberg/pull/54313#issuecomment-1716449966

LGTM!
